File: solutions/go/205.go
func isIsomorphic(s string, t string) bool {
	dict := make(map[byte]byte, len(s))
	used := make(map[byte]bool, len(t))

	for i := 0; i < len(s); i++ {
		if v, ok := dict[s[i]]; ok {
			if v != t[i] {
				return false
			}
		} else {
			if used[t[i]] {
				return false
			}

			dict[s[i]] = t[i]
			used[t[i]] = true
		}
	}

	return true
}

File: solutions/rust/205.rs
use std::collections::HashMap;

impl Solution {
    pub fn is_isomorphic(s: String, t: String) -> bool {
        let mut dict = HashMap::new();
        let mut used = HashMap::new();

        let s = s.into_bytes();
        let t = t.into_bytes();

        for i in (0..s.len()) {
            if let Some(v) = dict.get(&s[i]) {
                if *v != t[i] {
                    return false;
                }
            } else {
                if used.get(&t[i]).is_some() {
                    return false;
                }

                dict.insert(s[i], t[i]);
                used.insert(t[i], true);
            }
        }

        return true;
    }
}
